packages/odf/components/mcg/CreateObjectBucketClaim.tsx

```diff
@@ -51,6 +78,124 @@ export const CreateOBCForm: React.FC<CreateOBCFormProps> = (props) => {
     dispatch({ type: 'setProvisioner', name: sc?.provisioner });
   };
 
+  const generateOBCNameByNamespaceStore = (name: string): string => {
+    const base64Encoded = btoa(name).substring(0, 10);
+    return `destination-bucket-${name}-${base64Encoded}`.toLowerCase();
+  };
+
+  const generateBucketClassNameByNamespaceStore = (name: string): string => {
+    const base64Encoded = btoa(name).substring(0, 10);
+    return `generic-bucketclass-${name}-${base64Encoded}`.toLowerCase();
+  };
```

Comment:
prefix can be "destination-bucket" or "generic-bucketclass"
```suggestion
  const generateGenericName = (name: string, prefix: string): string => {
    const base64Encoded = btoa(name).substring(0, 10);
    return `${prefix}-${name}-${base64Encoded}`.toLowerCase();
  };
```